code/_onclick/hud/guardian.dm

```diff
@@ -14,7 +14,13 @@
 	using.screen_loc = ui_lhand
 	static_inventory += using
 
-	using = new /obj/screen/guardian/ToggleMode()
+	switch(owner.type)
+		if(/mob/living/simple_animal/hostile/guardian/ranged,/mob/living/simple_animal/hostile/guardian/healer)
+			using = new /obj/screen/guardian/ToggleMode()
+		if(/mob/living/simple_animal/hostile/guardian/assassin)
+			using = new /obj/screen/guardian/ToggleMode/Assassin()
+		else
+			using = new /obj/screen/guardian/ToggleMode/Inactive()
```

Comment:
Surely you could make this a var on holos and then replace this whole switch with `using = new toggle_path()`